packages/jest-cli/src/__tests__/TestRunner-test.js

```diff
@@ -21,3 +37,47 @@ test('.addReporter() .removeReporter()', () => {
   runner.removeReporter(SummaryReporter);
   expect(runner._dispatcher._reporters).not.toContain(reporter);
 });
+
+
+describe('_createInBandTestRun()', () => {
+  test('injects the rawModuleMap to each the worker in watch mode', () => {
+    const config = {watch: true};
+    const rawModuleMap = jest.fn();
+    const hasteContext = {moduleMap: {getRawModuleMap: () => rawModuleMap}};
+
+    const runner = new TestRunner(hasteContext, config, {maxWorkers: 2});
+
+    return runner._createParallelTestRun(
```

Comment:
Yeah I was thinking that as well. We could do that as a follow-up. TestRunner is one of the oldest parts of Jest and has been through a lot. I'm open to any ideas on how to improve the code to make it more testable. Feel free to create an issues to discuss or a PR to fix.

One thing that is untested as well is the performance test cache and that Jest runs failed tests first on a re-run. This all may break at any time. I'm really grateful you are adding such extensive unit tests for the watch mode, as it was entirely untested previously.